py/objstr.c

```diff
@@ -832,6 +832,10 @@ STATIC bool arg_looks_numeric(mp_obj_t arg) {
     ;
 }
 
+STATIC bool arg_looks_string(mp_obj_t arg) {
+    return MP_OBJ_IS_TYPE(arg, &mp_type_str) || MP_OBJ_IS_STR(arg);
```

Comment:
> If argument is not number, and format string does not have type field, and argument is string, then type should be set as string.

I don't think that's correct/general logic, and what I asked for is reference to CPython docs which say something like that. Just consider what will happen with an object which inherits from str? Which just has `__str__` special methods? You patch won't cover them, so it's a narrow-scope workaround, not a solution.